File: java_t3_full/Clone_2211.java
public static void main (String [] args) {
    int count = 0;
    String str = "hi this is is is line";
    String [] s1 = str.split (" ");
    for (int i = 0;
    i <= s1.length - 1; i ++) {
        if (s1 [i].equals ("is")) {
            count ++;
        }
    }
    System.out.println (count);
}


 public static void main (String [] args) {
    int count = 0;
    String str = "hi this is is is line";
    for (int i = 0; i < str.length(); i++) {
        if(str.substring(i, i+2).equalsIgnoreCase("is")) {
            count++;
        }
    }
    System.out.println (count);
}

File: cs_t4_full/Clone_2211.cs
internal static string GetIPAddress (HttpRequestBase request) {
    string forwarded = request.Headers ["Forwarded"];
    if (! String.IsNullOrEmpty (forwarded)) {
        foreach (string segment in forwarded.Split (',') [0].Split (';')) {
            string [] pair = segment.Trim ().Split ('=');
            if (pair.Length == 2 && pair [0].Equals ("for", StringComparison.OrdinalIgnoreCase)) {
                string ip = pair [1].Trim ('"');
                int left = ip.IndexOf ('['), right = ip.IndexOf (']');
                if (left == 0 && right > 0) {
                    return ip.Substring (1, right - 1);
                }
                int colon = ip.IndexOf (':');
                if (colon != - 1) {
                    return ip.Substring (0, colon);
                }
                return ip;
            }
        }
    }
    string xForwardedFor = request.Headers ["X-Forwarded-For"];
    if (! String.IsNullOrEmpty (xForwardedFor)) {
        return xForwardedFor.Split (',') [0];
    }
    return request.UserHostAddress;
}



internal static string GetIPAddress (HttpRequestBase request) {
    var getIpAddress = new Func<HttpRequestBase, string>(GetIpFromForwarded);
    var ipAddress = getIpAddress(request);
    if (ipAddress == null) {
        getIpAddress = GetIpFromXForwarded;
        ipAddress = getIpAddress(request);
    }
    return ipAddress ?? request.UserHostAddress;
}


